ethcontract-mock/src/details/mod.rs

```diff
@@ -93,6 +93,13 @@ impl MockTransport {
         address
     }
 
+    /// Deploys a new contract with the given ABI and address
+    pub fn deploy_with_address(&self, abi: &Abi, address: Address) {
+        let mut state = self.state.lock().unwrap();
+
+        state.contracts.insert(address, Contract::new(address, abi));
```

Comment:
Maybe we can assert that we aren't replacing an existing contract?

Reply:
Do you mean adding a simple assert like this:
`assert!(!state.contracts.contains_key(&address));`

Reply:
Also, `insert` returns the previous value, so you can also:
```suggestion
        assert!(
            state.contracts.insert(address, Contract::new(address, abi)).is_none(),
            "replacing contract at address {address:?}",
        );
```

Reply:
Added.